src/main/java/de/zalando/zmon/dataservice/proxies/kairosdb/KairosdbProxy.java

```diff
@@ -62,12 +67,18 @@ private void proxy(Request request, Writer writer, HttpServletResponse response)
         }
 
         try {
-            String data = executor.execute(request).returnContent().toString();
+            final String data = executor.execute(request).handleResponse(res -> {
+                final StatusLine status = res.getStatusLine();
+                HttpEntity entity = res.getEntity();
+                if (entity == null) {
+                    throw new ClientProtocolException("Response contains no content");
+                }
+                response.setStatus(status.getStatusCode());
+                final InputStreamReader reader = new InputStreamReader(entity.getContent());
```

Comment:
Correct me if I'm wrong, but here you just setting the original body and status code, headers are lost. Or it is a desired behaviour?
And what do you think, would it be reasonable to use `ProxyExchange` [https://cloud.spring.io/spring-cloud-gateway/1.0.x/multi/multi__building_a_gateway_using_spring_mvc.html](docs)? However, I have never worked with it, maybe it doesn't worth it.